portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +25,33 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private ArrayList<String> messages;
+
+  @Override
+  public void init() {
+    this.messages = new ArrayList<>();
```

Comment:
ImmutableList.of() is preferable.

You can add this to your pom.xml

 <dependency>
      <groupId>com.google.guava</groupId>
      <artifactId>guava</artifactId>
      <version>29.0-jre</version>
    </dependency>

Reply:
For my implementation of doPost in a later pull request, I appended to the ArrayList messages. In this case, would it be better to switch to ImmutableList and create a new list every time in doPost or continue using ArrayList?

Reply:
Given the List will grow, I think keeping it as an ArrayList is fine.

Reply:
Got it!